scripts/run_device_container.sh

```diff
@@ -8,6 +8,43 @@
 docker pull huntsmanarray/device_startup:latest
 
 #Run the docker container
+
 #Note that network=host to map container's ports to the hosts directly
-#This is OK for the pis etc. but probably not for the control computer
-docker run huntsmanarray/device_startup:latest --network host -it
+#This is OK for the pis etc. but probably not for the control computer.
+
+#-it allows the container to run in interactive mode.
+
+#--cap-add SYS_ADMIN --device /dev/fuse is necessary for the SSHFS mount.
+
+#-v ~/.ssh:/root/.ssh allows the container to use the credentials of the host
+#to setup ssh connections (i.e. the SSHFS). This is used for passwordless
+#login to the control computer, ***which must already be setup between the host
+#device and the control computer***.
+
+#docker run -it --network host --cap-add SYS_ADMIN --device /dev/fuse \
+#        -v ~/.ssh:/root/.ssh:ro huntsmanarray/device_startup:latest 
+#^Unable to set permissions properly this way 
+
+#Create a container 
+CONTAINER_ID=$(docker create -it --network host --cap-add SYS_ADMIN \
+        --device /dev/fuse huntsmanarray/device_startup:latest /bin/bash)
+
+#Start the container
+docker start $CONTAINER_ID
+
+#Extract some environment variables from the container
+CONTAINER_USER=$(docker exec $CONTAINER_ID echo $USER)
+CONTAINER_HOME=$(docker exec $CONTAINER_ID echo $HOME)
+CONTAINER_PANDIR=$(docker exec $CONTAINER_ID echo $PANDIR)
+
+#Copy the ssh credentials from the host to the container
+docker cp ~/.ssh $CONTAINER_ID:$CONTAINER_HOME/.ssh
+
+#Allow ssh credentials to be accessed by huntsman user
+docker exec -u root $CONTAINER_ID chown -R $CONTAINER_USER $CONTAINER_HOME/.ssh
+
+#Run the device script
+docker exec -it $CONTAINER_ID python $CONTAINER_PANDIR/huntsman-pocs/scripts/startup_device.py
+
+#Finally, kill the container
+docker exec -u root $CONTAINER_ID pkill bash
```

Comment:
We should try to use [`docker-compose`](https://docs.docker.com/compose/) for most of what this file does. It's more standard and more flexible. This doesn't apply to the last line but really the whole file.

Is there something complicated this script is trying to achieve that I don't see or is it just starting the container?

Reply:
The only reason I'm using exec is that, in order to share the ssh keys, it is required to know where to put them and which user requires them before hand. So this is a way of doing it without the script needing to assume environment variables inside the container. Will take a look at docker-compose.